### PR TITLE
Fix slow_conv_dilated{2,3}d backward unbatched grad_input indexing

### PR DESCRIPTION
**Pull request by cyyever**
Fixes silent corruption in `slow_conv_dilated{2,3}d_backward_{cpu,cuda}` on the unbatched code path.

  ### Root cause

  `slow_conv_dilated_all_{cpu,cuda}_template` iterates batch via `grad_input.select(0, elt)`. The four backward functions prepare an unsqueezed local `grad_input_`
  for the unbatched case but mistakenly pass `grad_input` (original shape) to the template. With a 3D input (2D conv) / 4D input (3D conv), `select(0, 0)` selects
  along the channel dim — writing one channel's worth of data into the wrong slot and leaving the other channels uninitialized.

  ### Fix

  Pass `grad_input_` to the template, matching how `input_` and `grad_output_` are already prepared. `grad_input_` is a view aliasing `grad_input`'s storage, so the
  returned `grad_input` (original shape) holds the correct data.

  Same pattern as the forward fix in #185352.

  ## Test Plan

  Added `test_slow_conv_dilated2d_unbatched` and extended `test_slow_conv_dilated3d_unbatched` to gradcheck the backward on the unbatched path with cudnn disabled.

  python test/nn/test_convolution.py 
    TestConvolutionNN.test_slow_conv_dilated2d_unbatched
    TestConvolutionNN.test_slow_conv_dilated3d_unbatched

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @aditew01

### CODE REVIEW

**Comment by Skylion007** [comment]
Inductor failures look real

**Comment by cyyever** [comment]
@Skylion007 fixed

File: test/nn/test_convolution.py [hunk 1]
```diff
@@ -3955,6 +4004,61 @@ def test_depthwise_conv_64bit_indexing(self, device):
...
+            dilation=[2, 2],
+        )
+        self._check_slow_conv_dilated(
+            torch.ops.aten.slow_conv_dilated2d, input, weight, bias, kwargs
+        )
+
+    @onlyNativeDeviceTypes
+    def test_slow_conv_dilated3d_unbatched(self, device):
+        if torch.device(device).type not in ("cpu", "cuda"):
+            self.skipTest("slow_conv_dilated3d only dispatches to CPU and CUDA")
```
Comment (Skylion007) [inline test/nn/test_convolution.py]: Can these be xfails instead?

File: test/nn/test_convolution.py [hunk 2]
```diff
@@ -3955,6 +4004,61 @@ def test_depthwise_conv_64bit_indexing(self, device):
...
+                    weight.detach().requires_grad_(),
+                    bias.detach().requires_grad_(),
+                )
+                torch.autograd.gradcheck(
+                    lambda i, w, b: op(i, w, bias=b, **kwargs), inputs
+                )
+
+    @onlyNativeDeviceTypes
+    def test_slow_conv_dilated2d_unbatched(self, device):
+        if torch.device(device).type not in ("cpu", "cuda"):
```
Comment (Skylion007) [inline test/nn/test_convolution.py]: Same make it xfails so we can enable test with new devices add support.

**Review by Skylion007 — APPROVED** [review]
